File: app/src/main/java/com/neome/java/api/meta/base/dto/DefnDtoLayoutCardFilter.java
// neome.ai API - do not change
//

package com.neome.java.api.meta.base.dto;

import com.neome.java.api.meta.base.Types.EnumDefnLayoutCardFilterKind;
import com.neome.java.api.meta.base.Types.EnumDefnSortOrder;
import com.neome.java.api.meta.base.Types.MetaIdField;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class DefnDtoLayoutCardFilter
{
  @Nullable
  public MetaIdField[] advanceFilterFieldIdSet;

  @Nullable
  public MetaIdField[] categoryFieldIdSet;

  @Nullable
  public EnumDefnLayoutCardFilterKind kind;

  @Nullable
  public Boolean showSearchBar;

  @Nullable
  public MetaIdField[] sortByFieldIdSet;

  @Nullable
  public EnumDefnSortOrder sortOrder;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnDtoLayoutCardFilter.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumDefnLayoutCardFilterKind
import com.neome.api.meta.base.Types.EnumDefnSortOrder
import com.neome.api.meta.base.Types.MetaIdField

interface DefnDtoLayoutCardFilter
{
  val advanceFilterFieldIdSet: List<MetaIdField>?
  val categoryFieldIdSet: List<MetaIdField>?
  val kind: EnumDefnLayoutCardFilterKind?
  val showSearchBar: Boolean?
  val sortByFieldIdSet: List<MetaIdField>?
  val sortOrder: EnumDefnSortOrder?
}